lib/graphql_schema.rb

```diff
@@ -24,6 +24,10 @@ def types
     @types ||= @hash.fetch('types').map{ |type_hash| TypeDefinition.new(type_hash) }.sort_by(&:name)
   end
 
+  def types_by_name
```

Comment:
A previous iteration of this method used would return `nil` if the type isn't found and now you are proposing the convenience method to raise a KeyError if the type isn't found.  I don't think it is clear what the behaviour should be for this convenience method and just using the hash will make it clearer what the behaviour is if the type isn't found.